### DISCUSSION TITLE
Option and Result types?

### DISCUSSION THREAD

**Discussion by StunxFS** [bait-lang/bait#145] [Questions and Answers] [answered]
Bait has option and result support?

**Comment by serkonda7** [comment, marked as answer]
Sadly Bait has no Option or Result types and no error handling yet.
However both types are planned to be implemented.

**Reply by StunxFS** [reply]
Ok, Bait seems like an interesting programming language, I could help you develop it too. 

Do you know the Rivet programming language? You might also be interested 🤔

**Reply by serkonda7** [reply]
Thanks, feel free to contribute if you like. I'm happy to assist with any questions you might have.

Rivet looks interesting too. I'm looking forward to the selfhosted version!

**Reply by StunxFS** [reply]
Thanks 😁, and yes, Rivet will soon have a 100% functional self-hosted compiler, I just need to implement a codegen and that's it 🥳

**Reply by serkonda7** [reply]
Hi, I started working towards Result type a few days ago.
So it will be implemented soonish :)

Option types remain on the todo list tho

**Comment by serkonda7** [comment]
With the merge of https://github.com/tiabeast/bait/pull/172, Result types and error handling is now merged :tada: 

It might still contain some bugs an edge cases but is a good starting point.

Option types are not done but won't be prioritized for now as their functionality can be replicated using Result.

**Reply by StunxFS** [reply]
Perfect! 🥳
